src/lib/groovy/ModsDotGroovy.groovy

```diff
@@ -380,6 +380,37 @@ class ModsDotGroovy {
             }
         }
 
+        final Matcher mrMatcher = modInfo.displayUrl =~ $/.*modrinth.com/mod/([^?/]+)/?/$
+        println "displayUrl: ${modInfo.displayUrl}"
+        println "matches? ${mrMatcher.matches()}"
+        if (mrMatcher.matches()) {
+            httpClient ?= HttpClient.newBuilder().build()
+            final String updateJsonUrl = "https://api.modrinth.com/updates/${mrMatcher.group(1)}/forge_updates.json"
+
+            // modrinth's api doesn't factor in the actual mod jar's mods.toml version - just the published version,
+            // so we need to check that the returned promos map contains the mod's current version to avoid constantly
```

Comment:
If we haven't published the mod yet, it won't include the current version, right? Unless I'm missing something.

Reply:
You're right, I seemed to have overlooked that, although I can't think of another way of avoiding trouble.

Unlike CurseUpdate, Modrinth's API doesn't actually look at any of the mods.toml files inside jars. When versions mismatch, best case it'll waste bandwidth for all users and never notify for updates, worst case it'll always nag them that there's a new version available even when they're on the latest.

Modrinth seems to be aware of this as they have a big orange caution message in the doc link you sent, although this requires devs to workaround the issue which isn't ideal for multiplatform mods where it's reasonable to want your version format published on the site to mention the loader but don't want it in your mods.toml so that more accurate version comparisons can be made.

I think the best approach would be to leave devs to opt-into modrinth's updateJsonUrl manually as usual rather than risk quietly causing the issues mentioned.

Reply:
Reverted following discussion on MMD #groovy-mc